projects/vdk-core/src/vdk/internal/builtin_plugins/connection/database_managed_connection.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2023-2024 Broadcom
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Container
+from typing import Optional
+
+from vdk.internal.builtin_plugins.connection.decoration_cursor import DecorationCursor
+from vdk.internal.builtin_plugins.connection.execution_cursor import ExecuteOperationResult
+from vdk.internal.builtin_plugins.connection.execution_cursor import ExecutionCursor
+from vdk.internal.builtin_plugins.connection.recovery_cursor import RecoveryCursor
+
+
+class IDatabaseManagedConnection:
```

Comment:
Not sure if we need a separate interface for these. They can be part of IManagedConnection instead.

Reply:
But there becomes a circular dependency since we use the ManagedCursor within ManagedConnectionBase

Reply:
Sorry, didn't think about that. It's fine to leave it as is if you get a circular dependency.